### PR TITLE
[#IC-254] New Pagopa proxy config and specs

### PR DESCRIPTION
**Pull request by BurnedMarshal**
#### List of Changes

Updated `pagopa-proxy` OpenAPI specifications with a temporary file (when the production specs will be available over pagopa/io-pagopa-proxy#335 the local specs will be deleted)
Generate a new client with the required header for authorization
Refactor to handle the new generated client
Added 2 new ENV variables to read the API key for production and UAT

#### Motivation and Context

The instance of PagoPA Proxy will be migrated under another subscription. The Authentication will change so we need to update the API specification and use properly the new generated client.

#### How Has This Been Tested?

unit test
#### Types of changes

- [X] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: src/controllers/__tests__/pagoPAProxyController.test.ts [hunk 1]
```diff
@@ -85,7 +85,9 @@ describe("PagoPAProxyController#getPaymentInfo", () => {
 
     const pagoPAClientFactory = new PagoPAClientFactory(
       process.env.PAGOPA_API_URL_PROD as string,
-      process.env.PAGOPA_API_URL_TEST as string
+      process.env.PAGOPA_API_KEY_PROD as string,
```
Comment (michaeldisaro) [inline src/controllers/__tests__/pagoPAProxyController.test.ts:88]: can we extract this to avoid repeating the code?

**Review by michaeldisaro — APPROVED** [review]
Apart a minor change code LGTM.

File: src/services/pagoPAProxyService.ts [hunk 2]
```diff
@@ -82,7 +82,7 @@ export default class PagoPAProxyService {
         isTest ? PagoPAEnvironment.TEST : PagoPAEnvironment.PRODUCTION
       );
       const validated = await client.activatePayment({
-        body: paymentActivationsPostRequest
+        paymentActivationsPostRequest
```
Comment (balanza) [inline src/services/pagoPAProxyService.ts:85]: why this change?

File: src/controllers/__tests__/pagoPAProxyController.test.ts (hunk 1 above)
Comment (BurnedMarshal) [reply src/controllers/__tests__/pagoPAProxyController.test.ts:88]: Sorry, I don't understand what you mean with avoid repeating code. Can you provide me an example?

Comment (michaeldisaro) [reply src/controllers/__tests__/pagoPAProxyController.test.ts:88]: Yes, sorry, I pressed + and github commented just the line. I mean extracting this code outside the test cases:
```
const pagoPAClientFactory = new PagoPAClientFactory(
      process.env.PAGOPA_API_URL_PROD as string,
      process.env.PAGOPA_API_KEY_PROD as string,
      process.env.PAGOPA_API_URL_TEST as string,
      process.env.PAGOPA_API_KEY_UAT as string
    );
const pagoPAProxyService = new PagoPAProxyService(pagoPAClientFactory);
const controller = new PagoPAProxyController(pagoPAProxyService);
 ```

From what I see it seems it never changes.

Comment (BurnedMarshal) [reply src/controllers/__tests__/pagoPAProxyController.test.ts]: Fixed in [e5790f1](https://github.com/pagopa/io-backend/pull/874/commits/e5790f124d6f4beb72fbb0c64c847aa45db1ba4f)

**Review by michaeldisaro — APPROVED** [review]
